src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -576,7 +567,49 @@ int fetch_object_timeout_handler(event_loop *loop, timer_id id, void *context) {
              end_time - start_time);
   }
 
-  return LOCAL_SCHEDULER_FETCH_TIMEOUT_MILLISECONDS;
+  /* Wait at least LOCAL_SCHEDULER_FETCH_TIMEOUT_MILLISECONDS before running
+   * this timeout handler again. But if we're waiting for a large number of
+   * objects, wait longer (e.g., 10 seconds for one million objects) so that we
+   * don't overwhelm the plasma manager. */
+  return std::max(kLocalSchedulerFetchTimeoutMilliseconds,
+                  int64_t(0.01 * num_object_ids));
+}
+
+static int64_t reconstruct_counter = 0;
```

Comment:
add a todo that this should become a class member variable for the C++ refactor (or make it a member variable in LocalSchedulerState)